src/actions/editor.js

```diff
@@ -323,24 +295,32 @@ const executeSendRequest = (formValues, contentLanguages, user, updateExisting,
                 }
 
                 if(response.status === 200 || response.status === 201) {
-                    // create sub events after creaing super event successfully
+                    // create or update sub events after creating/updating super event successfully
                     if (json.super_event_type === 'recurring') {
-                        const formWithAllSubEvents = combineSubEventsFromEditor(formValues, updateExisting)
-                        dispatch(
-                            sendRecurringData(formWithAllSubEvents, contentLanguages, user, updateExisting, publicationStatus, json['@id'])
-                        )
-                        dispatch(sendDataComplete(json, actionName))
+                        // update sub events
+                        if (updateExisting) {
+                            const subEventsToUpdate = getState().subEvents.items
+                            dispatch(updateRecurringData(preparedFormValues, subEventsToUpdate, contentLanguages, user, json.id))
```

Comment:
Hm, is there some reason that `updateRecurringData` and `sendRecurringData` have such differing signatures? I would rather refactor one or the other so that they behave similarly.